03week/ticTacToe.js

```diff
@@ -1,93 +1,158 @@
-'use strict';
+"use strict";
 
-const assert = require('assert');
-const readline = require('readline');
+const assert = require("assert");
+const readline = require("readline");
 const rl = readline.createInterface({
   input: process.stdin,
   output: process.stdout
 });
 let board = [
-  [' ', ' ', ' '],
-  [' ', ' ', ' '],
-  [' ', ' ', ' ']
+  [" ", " ", " "],
+  [" ", " ", " "],
+  [" ", " ", " "]
 ];
 
-let playerTurn = 'X';
+let playerTurn = "X";
 
 function printBoard() {
-  console.log('   0  1  2');
-  console.log('0 ' + board[0].join(' | '));
-  console.log('  ---------');
-  console.log('1 ' + board[1].join(' | '));
-  console.log('  ---------');
-  console.log('2 ' + board[2].join(' | '));
+  console.log("   0  1  2");
+  console.log("0 " + board[0].join(" | "));
+  console.log("  ---------");
+  console.log("1 " + board[1].join(" | "));
+  console.log("  ---------");
+  console.log("2 " + board[2].join(" | "));
 }
 
 function horizontalWin() {
   // Your code here
+  for (let i = 0; i <= 2; i++) {
+    if (
+      board[i][0] == board[i][1] &&
+      board[i][1] == board[i][2] &&
+      board[i][0] != " "
+    ) {
+      return true;
+    }
+  }
+  return false;
 }
 
 function verticalWin() {
   // Your code here
+  for (let i = 0; i <= 2; i++) {
+    if (
+      board[0][i] == board[1][i] &&
+      board[1][i] == board[2][i] &&
+      board[0][i] != " "
+    ) {
+      return true;
+    }
+  }
+  return false;
 }
 
 function diagonalWin() {
   // Your code here
+  if (
+    (board[0][0] == board[1][1] && board[1][1] == board[2][2]) ||
```

Comment:
This code will say its a diagonal win, if they are all empty since they will be equal to each other.